src/web.ts

```diff
@@ -1,22 +1,86 @@
-import { WebPlugin } from '@capacitor/core';
-import { SignInWithApplePlugin } from './definitions';
+import { WebPlugin } from "@capacitor/core";
+
+import {
+  SignInWithApplePlugin,
+  InitOptions,
+  SignInResponse,
+  SignInError,
+} from "./definitions";
+
+declare var window: any;
+
+export class SignInWithAppleWeb extends WebPlugin
+  implements SignInWithApplePlugin {
+  hasInitialized = false;
 
-export class SignInWithAppleWeb extends WebPlugin implements SignInWithApplePlugin {
   constructor() {
     super({
-      name: 'SignInWithApple',
-      platforms: ['web']
+      name: "SignInWithApple",
+      platforms: ["web"],
+    });
+  }
+
+  async Init(options: InitOptions): Promise<void> {
+    this.loadAppleScript(() => {
+      if (window && window.AppleID && !this.hasInitialized) {
+        const { clientId, scope, redirectURI, state, usePopup } = options;
+
+        window.AppleID.auth.init({
+          clientId: clientId,
+          scope: scope,
+          redirectURI: redirectURI,
+          state: state,
+          usePopup: usePopup !== undefined ? usePopup : false,
+        });
+
+        this.hasInitialized = true;
+      }
     });
   }
 
-  async Authorize(): Promise<{response: any}> {
-    return;
+  Authorize(): Promise<SignInResponse> {
+    return new Promise<SignInResponse>((resolve, reject) => {
+      const buildReject = (error: string) => reject({ error } as SignInError);
+      const wait = (callback: Function) => setTimeout(callback, 100);
```

Comment:
I definitely wouldn't use an arbitrary timeout here, that's got a high likelyhood of failing at some point. Perhaps use a read-resolver pattern instead, maybe with a `setInterval`?